src/main/java/org/codeforamerica/shiba/application/parsers/ApplicationDataParser.java

```diff
@@ -11,11 +12,11 @@ public ApplicationDataParser(ParsingConfiguration parsingConfiguration) {
 
     public abstract T parse(ApplicationData applicationData);
 
-    protected static double getDouble(ApplicationData applicationData, PageInputCoordinates pageInputCoordinates) {
+    protected static Money getMoney(ApplicationData applicationData, PageInputCoordinates pageInputCoordinates) {
         try {
-            return Double.parseDouble(applicationData.getValue(pageInputCoordinates).replace(",",""));
+            return new Money(Double.parseDouble(applicationData.getValue(pageInputCoordinates).replace(",","")));
```

Comment:
Looks like we could add a `Money(String val)` constructor and reference it here since the `Money(double val)` seems to turn it into a string anyways

Reply:
One nice thing about keeping it only as a `double` is it forces you to make a parseable number in the constructor.. but parsing `double`s as a String to use in a `BigDecimal` guarantees that it can be parsed since it'd have to be written as a number beore that point. [Here's an article](https://cdn.app.compendium.com/uploads/user/e7c690e8-6ff9-102a-ac6d-e4aebca50425/34d3828b-c12e-4c7e-8942-b6b7deb02e12/File/8712e504216fcf1c3b5f37402c013172/four_common_pitfalls_of_the_bigdecimal_class_and_how_to_avoid_them.pdf) I was using to try to make the call.. it wasn't conclusive. cc: @andrewedstrom

Reply:
hmI definitely can see the argument for that. Do y'all wanna jump on a call to discuss options?

Reply:
Sure - during break session. 

> parsing doubles as a String to use in a BigDecimal guarantees that it can be parsed since it'd have to be written as a number beore that point

I think Im misunderstanding this part. We get a String input, parse it to double for the constructor and then back to String within the constructor. The BigDecimal constr will also check if it is a valid decimal. Is there an example where it would break if we just passed the String directly to the BigDecimal?

Reply:
The changes I just made (which consolidated around the String constructor) should make it work in most cases we can predict since we're stripping commas and use the Money Validation.